static/js/unread_ops.js

```diff
@@ -29,6 +29,21 @@ export function mark_all_as_read() {
                 save_compose: true,
             });
         },
+        error(xhr) {
+            if (xhr.status === 502) {
+                /* When trying to mark a lot of messages as read, it's
+                   possible for the request to the server to
+                   time out after making some progress.
+                   The same behavior makes sense as in the case of success,
+                   reload the client. */
```

Comment:
I don't think this is a pattern we should have at all -- see https://github.com/zulip/zulip/pull/22988/commits/10dbe6fcd851fde01fbbdccd8967bd15cd940da9#r986351526.

_Anything_ which says "when the server is overloaded, try harder" is going to bite us.  We should have an explicit server code for "we completed part, please continue."  Forcing a full re-fetch of `/`, _which is the most heavy-weight thing we can do_, when the server is giving 502s to try to shed load, is not the right pattern.

Reply:
Is there an nginx way (and would it be reasonable to use for this?) to make an endpoint-specific timeout for this with a different response code?

Reply:
It's certainly an option to do a 45s timeout at the Python level for this endpoint, like we do in the markdown processor, and have that return a special error code.

Reply:
Okay, I got started on adding that. Any suggestions on what the response code should be? Tentatively i went with `503` but i dont know if we want this to be a `5xx` at all? Overall i can't find a response code that would be reasonably close to meaning "your request timed out, but just retry it".

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/504 is the best match I can find, I think?

Reply:
Ah, that looks better than  503, ill go with that for now. @alexmv Any thoughts on this?

Reply:
A 504 or 503 could still be sent from an intermediate proxy (like nginx or AWS' ALB) and sending it during normal operation still seems confusing.  504 and 503 are both 5xx responses, which I believe we shouldn't be sending for normal, unexceptional behavior.

I would go with 200, with JSON content that says that we truncated the data.  Or possibly 206 if we really want to differentiate based on status code.

Fundamentally, there is no "server failure" in this case, which is what 5xx is for.  We have made an decision to limit the amount of time that we'll spend on such requests (which we should be sure to put in the API docs, along with whatever status code we end up with ereh), but each one isn't _failing_.